text/0000-fn_name_macro.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: fn_name_macro
+- Start Date: 2019-11-14
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+<!-- https://github.com/rust-lang/rfcs/issues/1743 -->
+
+# Summary
+[summary]: #summary
+
+This RFC adds an additional macro, `function!`, to the `core` crate. When invoked, the macro expands to the name of the function that contains the call site.
+
+# Motivation
+[motivation]: #motivation
```

Comment:
Some elaboration on the common use case would be good. In the issue you mention:
> I wouldn't normally think to pair func_name!() with line!() and file!() because usually when i want it I want "emergency println debugging" and i'm just throwing in a few printers and i delete them once i've sorted out what's wrong.

and I'm wondering whether you expect the macro to be used in e.g. `dbg!(...)` (which does print out `file!()` and `line!()`).

Reply:
Good point. I will add an example usage to the guide level explanation.